plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3284,16 +3284,22 @@ public void testVoidTransform()
         assertUpdate("INSERT INTO test_void_transform " + values, 7);
         assertQuery("SELECT * FROM test_void_transform", values);
 
-        assertQuery("SELECT COUNT(*) FROM \"test_void_transform$partitions\"", "SELECT 1");
-        if (format != AVRO) {
+        MaterializedResult partitions = computeActual("SELECT * FROM \"test_void_transform$partitions\"");
+        assertThat(partitions.getRowCount()).isEqualTo(1);
+        if (format == AVRO) {
+            assertQuery(
+                    "SELECT partition.d_null, record_count, file_count, data.d.min, data.d.max, data.d.null_count, data.d.nan_count, data.b.min, data.b.max, data.b.null_count, data.b.nan_count FROM \"test_void_transform$partitions\"",
+                    "VALUES (NULL, 7, 1, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL)");
+        }
+        else if (partitions.getColumnNames().contains("partition")) {
```

Comment:
Why don't we know whether `partitions.getColumnNames()`?

Reply:
This is due to https://github.com/apache/iceberg/pull/3059 but for some reason it only affects the REST catalog. I don't understand why and it's hard to debug since the behavior is inside the REST server.

Reply:
cc @danielcweeks @nastra

Reply:
@electrum can this hack be guarded by "are we using REST catalog" and also be adorned with a code comment explaining why we need the hack at all?

Reply:
I'm not sure why this would only affect the REST catalog. Isn't the underlying issue as described in https://github.com/trinodb/trino/pull/15726#discussion_r1082243472, where the test is trying to query a partition that has a void partition but after https://github.com/apache/iceberg/pull/3059 this is considered as unpartitioned?

Reply:
Looks like we can revert this after #16588